src/stateManager/StateManager.ts

```diff
@@ -278,122 +273,80 @@ class StateManager {
+    public onSignedBlock(
+        signedBlock: SignedBlockStruct
     ): Promise<ExecutionFlags> {
-        // Default everything to SUCCESS + no AgreementFlag
-        let finalExecutionFlag: ExecutionFlags = ExecutionFlags.SUCCESS;
-        let finalAgreementFlag: AgreementFlag | undefined = undefined;
-        const block = blk ?? Block.decode(signedBlock.encodedBlock);
-
-        try {
-            await this.mutex.lock();
-            const result = await this.validationService.validateSignedBlock(
-                signedBlock,
-                block
-            );
-
-            finalExecutionFlag = result.flag;
-            finalAgreementFlag = result.agreementFlag;
-
-            // If validation passed, apply the transaction
-            if (result.success && result.flag === ExecutionFlags.SUCCESS) {
-                const applyResult = await this.applyTransaction(
-                    block.transaction
-                );
-                if (!applyResult.success) {
-                    finalExecutionFlag = ExecutionFlags.DISPUTE;
-                    finalAgreementFlag = AgreementFlag.INCORRECT_DATA;
-                } else {
-                    // store exit block, exit points, state snapshot and the block
-                    this.storeExitChannelBlock(
-                        applyResult.exitChannels,
-                        block.coordinates
-                    );
-
-                    this.handleStateSnapshotStorage(
-                        applyResult.encodedState,
-                        block.forkId
-                    );
-
-                    this.storage.blocks.storeBlock(signedBlock);
-
-                    // Schedule the success callback
-                    scheduleTask(
-                        applyResult.successCallback,
-                        0,
-                        "stateTransitionSuccessCallback"
-                    );
-                }
-            }
-
-            return finalExecutionFlag;
-        } finally {
-            // Safety check: must have an execution flag
-            if (finalExecutionFlag === undefined) {
-                throw new Error(
-                    "StateManager - onSignedBlock - Internal Error - flag undefined"
-                );
-            }
-
-            // Process the final decision
-            await this.executionDecisionProcessor.process(
-                signedBlock,
-                finalExecutionFlag,
-                finalAgreementFlag
-            );
-            this.mutex.unlock();
-        }
+        return this.onBlockConfirmation({
+            signedBlock,
+            signatures: []
+        });
     }
 
     // Passes the block confirmation through a verification pipeline and returns an execution flag
     public async onBlockConfirmation(
-        signedBlock: SignedBlockStruct,
-        confirmationSignature: Signature,
-        block?: Block
+        blockConfirmation: BlockConfirmationStruct
     ): Promise<ExecutionFlags> {
```

Comment:
Take a look at the right part of the diagram where it explains the execution flags meaning and the conclusion - we don't have to return the execution flag just a flag `shouldDisconnect` and the flags can be removed all together and substituted with functions `success(), dispute(), notReady() ...` that do the common operation of those functions respectively - if for some reason you still need `ExecutionFlags` internally within `StateManager` sure, but in general they're not needed